python_src/src/lamp_py/migrations/versions/performance_manager_staging/005_96187da84955_remove_metadata.py

```diff
@@ -0,0 +1,102 @@
+"""remove_metadata
+
+Revision ID: 96187da84955
+Revises: 45dedc21086e
+Create Date: 2023-12-28 12:18:25.412282
+
+check that all information in the metadata table has been copied to the
+metadata database before dropping the table and its indexes entirely.
+"""
+from alembic import op
+from sqlalchemy.dialects import postgresql
+from sqlalchemy.exc import ProgrammingError
+import logging
+import sqlalchemy as sa
+
+from lamp_py.postgres.postgres_utils import DatabaseIndex, DatabaseManager
+from lamp_py.postgres.metadata_schema import MetadataLog
+
+# revision identifiers, used by Alembic.
+revision = "96187da84955"
+down_revision = "45dedc21086e"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    while True:
+        try:
+            rpm_db_manager = DatabaseManager(
+                db_index=DatabaseIndex.RAIL_PERFORMANCE_MANAGER
+            )
+            md_db_manager = DatabaseManager(db_index=DatabaseIndex.METADATA)
+
+            with rpm_db_manager.session.begin() as session:
+                legacy_result = session.execute("SELECT path FROM metadata_log")
+                legacy_paths = set(
+                    [record[0] for record in legacy_result.fetchall()]
+                )
+
+        except ProgrammingError as error:
+            # Error 42P01 is an 'Undefined Table' error. This occurs when there is
+            # no metadata_log table in the rail performance manager database
+            #
+            # Raise all other sql errors
+            if error.orig.pgcode == "42P01":
+                logging.info("No Metadata Table in Rail Performance Manager")
+                legacy_paths = set()
+            else:
+                raise
+
+        modern_result = md_db_manager.select_as_list(
+            sa.select(MetadataLog.path)
+        )
+        modern_paths = set([record["path"] for record in modern_result])
+
+        missing_paths = legacy_paths - modern_paths
+        if len(missing_paths) == 0:
+            break
+        else:
+            logging.error(
+                "Detected %s paths in Legacy Metadata Table not found in Metadata Database",
+                len(missing_paths),
+            )
```

Comment:
If the Metadata RDS tables have not been created before this block runs, this query will throw and kill the ECS. 

One suggestion:
```suggestion
        try:
            modern_result = md_db_manager.select_as_list(
                sa.select(MetadataLog.path)
            )
            modern_paths = set([record["path"] for record in modern_result])
      
            missing_paths = legacy_paths - modern_paths
            if len(missing_paths) == 0:
                break
            else:
                logging.error(
                    "Detected %s paths in Legacy Metadata Table not found in Metadata Database",
                    len(missing_paths),
                )
            except Exception as _:
                time.sleep(15)
                continue
```